score.py

```diff
@@ -128,6 +134,95 @@ def run_inference(images_with_header: List[ImageWithHeader],
+            # try to extract only DICOM files, but these need not have
+            # an extension
+            if not zipped_file.is_dir() and (
+                    zipped_file.filename.endswith('.dcm') or '.' not in zipped_file.filename):
+                # discard the path, if any, to just get the filename and suffix
+                zipped_file.filename = os.path.basename(zipped_file.filename)
+                zip_file.extract(zipped_file, str(extraction_folder))
+
+
+def convert_zipped_dicom_to_nifti(test_image: Path, model_folder: Path) -> Tuple[Path, Path]:
+    """
+    Given a zip file, extract DICOM series and convert to Nifti format.
+
+    :param test_image: Path to a zip file.
+    :param model_folder: Scratch folder to extract files into.
+    :return: Pair of path to Nifti file and path to folder of reference series.
+    """
+    reference_series_folder = model_folder / "temp_extraction"
+    extract_zipped_dicom_series(test_image, reference_series_folder)
+    nifti_filename = model_folder / "temp_nifti.nii.gz"
+    load_dicom_series_and_save(reference_series_folder, nifti_filename)
+    return nifti_filename, reference_series_folder
+
+
+def convert_rgb_colour_to_hex(colour: TupleInt3) -> str:
+    """
+    Config colours are stored as TupleInt3's, but DICOM-RT convert expects
+    hexadecimal strings. This function converts them into the correct
+    format.
+
+    :param colour: RGB colour as a TupleInt3.
+    :return: Colour formatted as a hex string.
+    """
+    return '{0:02X}{1:02X}{2:02X}'.format(colour[0], colour[1], colour[2])
+
+
+def convert_nifti_to_zipped_dicom_rt(nifti_file: Path, reference_series: Path, model_folder: Path,
+                                     config: SegmentationModelBase) -> Path:
+    """
+    Given a Nifti file and a reference DICON series, convert Nifti file to DICOM-RT file.
+
+    :param nifti_file: Path to Nifti file.
+    :param reference_series: Path to folder containing reference DICOM series.
+    :param model_folder: Scratch folder to extract files into.
+    :param config: Model config.
+    :return: Path to DICOM-RT file.
+    """
+    dicom_rt_file = model_folder / nifti_file.with_suffix(".dcm").name
+    (stdout, stderr) = rtconvert(
+        in_file=nifti_file,
+        reference_series=reference_series,
+        out_file=dicom_rt_file,
+        struct_names=config.ground_truth_ids_display_names,
+        struct_colors=[convert_rgb_colour_to_hex(rgb) for rgb in config.colours],
+        fill_holes=config.fill_holes)
+    logging.debug("stdout: %s", stdout)
+    logging.debug("stderr: %s", stderr)
+    dicom_rt_zip_file = dicom_rt_file.with_suffix(dicom_rt_file.suffix + '.zip')
+    with zipfile.ZipFile(dicom_rt_zip_file, 'w') as dicom_rt_zip:
+        dicom_rt_zip.write(dicom_rt_file, dicom_rt_file.name)
+    return dicom_rt_zip_file
+
+
+def check_input_file(data_folder: Path, file: str) -> Path:
+    """
+    Check that input file exists, and raise an exception if not.
```

Comment:
The docu does not say much beyond what the function name would let me guess (and I consider such documentation a nuisance). In particular, docu does not say how the two arguments interact?